model-server/src/test/kotlin/org/modelix/model/server/handlers/ModelReplicationServerTest.kt

```diff
@@ -440,4 +450,110 @@ class ModelReplicationServerTest {
             )
         }
     }
+
+    @Test
+    fun `putRepositoryObjects detects missing values`() = runWithTestModelServer { _, _ ->
+        val client = createClient { install(ContentNegotiation) { json() } }
+        val modelClient = ModelClientV2.builder().url("v2").client(client).build().also { it.init() }
+        val repositoryId = RepositoryId("repo1")
+        modelClient.initRepository(repositoryId)
+        val invalidRequestBody = "7-m7S*flQz_CNc9F3ipSQ5Iz7SZJSyn2r0_1PPVvh8yA"
+
+        val response = client.put {
+            url {
+                takeFrom("v2")
+                appendPathSegments("repositories", repositoryId.id, "objects")
+            }
+            contentType(ContentType.Text.Plain)
+            setBody(invalidRequestBody)
+        }
+
+        val expectedProblem = Problem(
+            type = "/problems/uploaded-object-key-without-object-value",
+            title = "Uploaded object key without object value.",
+            status = 400,
+            detail = "Uploaded object key `7-m7S*flQz_CNc9F3ipSQ5Iz7SZJSyn2r0_1PPVvh8yA` without object value.",
+        )
```

Comment:
I'm not sure if it is a good idea to test all aspects of the generated response. For instance, the detail is meant as a human-readable explanation. If we later decide that the details need a slight rewording to be better understandable, we'd have a million of tests to update. So I don't see much benefit in comparing this string but some drawbacks.

Reply:
btw, a we now have kotest, you could use `shouldBeEqualIgnoringFields` to achieve such a comparison.

Reply:
I think user messages should also be tested, because they can break.
And rewording will not happen often.

Reply:
We could move all of this to a single test for the exception class, asserting about the generated detail message and ignore the rest here. That way, when the next test asserts about the 400 response here, we at least don't have two spots checking the exact same detail message generation pattern.